mods/ctf/ctf_map/ctf_traps.lua

```diff
@@ -101,3 +101,107 @@ minetest.register_node("ctf_map:reinforced_cobble", {
 	groups = {cracky = 1, stone = 2},
 	sounds = default.node_sound_stone_defaults(),
 })
+
+minetest.register_node("ctf_map:landmine", {
+	description = "Landmine",
+	drawtype = "nodebox",
+	tiles = {"ctf_map_landmine.png", "ctf_map_landmine.png^[transformFY"},
+	inventory_image = "ctf_map_landmine.png",
+	paramtype = "light",
+	sunlight_propagates = true,
+	walkable = false,
+	groups = {cracky=1, level=2},
+	node_box = {
+		type = "fixed",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.4, 0.5},
+	},
+	selection_box = {
+		type = "fixed",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.4, 0.5},
+	},
+	after_place_node = function(pos, placer, itemstack, pointed_thing)
+		local meta = minetest.get_meta(pos)
+		local name = placer:get_player_name()
+
+		meta:set_string("placer", minetest.serialize({
+			name = name,
+		}))
+	end
+})
+
+minetest.register_abm({
+	label = "Landmine",
+	nodenames = {"ctf_map:landmine"},
+	interval = 0.5,
+	chance = 1,
+	action = function(pos, node, active_object_count, active_object_count_wider)
+		local players = minetest.get_objects_inside_radius(pos, 2)
+		local to_be_damaged = {}
+		local i = 1
+		local meta = minetest.get_meta(pos)
+		local placer = minetest.deserialize(meta:get_string("placer"))
+		local placer_team = ctf_teams.get(placer.name) 
+		local placer_obj = placer and minetest.get_player_by_name(placer.name)
+		
+		for _, v in pairs(players) do
+			if v:is_player() and ctf_teams.get(v:get_player_name()) ~= placer_team then
```

Comment:
Players have the ability to change teams, so the landmine shouldn't pose a threat to the team it was originally placed for if the player happens to switch. Untested, but this should get the gist of my idea down.
```suggestion
	after_place_node = function(pos, placer, itemstack, pointed_thing)
		local meta = minetest.get_meta(pos)
		local name = placer:get_player_name()

		meta:set_string("placer", minetest.serialize({
			name = name,
			team = ctf_teams.get(name)
		})
	end
})

minetest.register_abm({
	label = "Landmine",
	nodenames = {"ctf_map:landmine"},
	interval = 0.5,
	chance = 1,
	action = function(pos, node, active_object_count, active_object_count_wider)
		local players = minetest.get_objects_inside_radius(pos, 2)
		local to_be_damaged = {}
		local i = 1
		local meta = minetest.get_meta(pos)
		local placer = minetest.deserialize(meta:get_string("placer"))
		local placer_team = placer.team
		local placer_obj = placer and minetest.get_player_by_name(placer.name)
		
		for _, v in pairs(players) do
			if v:is_player() and ctf_teams.get(v:get_player_name()) ~= placer_team then
```

Reply:
Hmm. Waiting for the others' comments. But I think it is a good idea to store placer's team and then check for it

Reply:
Just use the code from damage_cobble, don't make it by yourself.